### ISSUE TITLE
revoked messages

### ISSUE THREAD

**Issue by m00nty** [Schwartzblat/WhatsAppPatcher#29]
Revoked messages get deleted I don't think the patch is working.

**Comment by jeverling** [comment]
Same for me, with version WhatsApp version ` 2.23.17.15`.

**Comment by jeverling** [comment]
BTW, the latest stable apk can also be downloaded from https://www.whatsapp.com/android/

**Comment by Schwartzblat** [comment]
I am an IPhone user so it's kind of hard for me to test some of the changes :)
Please let me know if the problem was fixed in this update @m00nty @jeverling .

**Comment by jeverling** [comment]
Thanks a lot Alon! Will try a bit later today and get back.

**Comment by jeverling** [comment]
Hi Alon, unfortunately the latest patched version you attached to release 1.0.9 crashes for me.
It installs without issues, but when trying to open the app it shows the splash screen (_from Meta_) and then reproducibly crashes.
Here is some logcat output, in case that helps:

```
08-25 16:11:17.357 17517 17517 E AndroidRuntime: FATAL EXCEPTION: main
08-25 16:11:17.357 17517 17517 E AndroidRuntime: Process: com.whatsapp, PID: 17517
08-25 16:11:17.357 17517 17517 E AndroidRuntime: java.lang.VerifyError: Verifier rejected class X.2pB: boolean X.2pB.A04(X.33c, boolean) failed to verify: boolean X.2pB.A04(X.33c, boolean): [0x2] return-void not expected (declaration of 'X.2pB' appears in /data/app/com.whatsapp-yWKLN7ROM94bK0XwNFSRGA==/base.apk:classes2.dex)
08-25 16:11:17.357 17517 17517 E AndroidRuntime:        at X.3aZ.A2M(Unknown Source:5427)
08-25 16:11:17.357 17517 17517 E AndroidRuntime:        at X.3aZ.get(Unknown Source:55)
08-25 16:11:17.357 17517 17517 E AndroidRuntime:        at X.3ab.get(:595754)
08-25 16:11:17.357 17517 17517 E AndroidRuntime:        at X.3aZ.A2M(Unknown Source:2376)
08-25 16:11:17.357 17517 17517 E AndroidRuntime:        at X.3aZ.get(Unknown Source:55)
08-25 16:11:17.357 17517 17517 E AndroidRuntime:        at X.3ab.get(:595754)
08-25 16:11:17.357 17517 17517 E AndroidRuntime:        at X.3aZ.A2M(Unknown Source:2502)
08-25 16:11:17.357 17517 17517 E AndroidRuntime:        at X.3aZ.get(Unknown Source:55)
08-25 16:11:17.357 17517 17517 E AndroidRuntime:        at X.3ab.get(:595754)
08-25 16:11:17.357 17517 17517 E AndroidRuntime:        at X.3aZ.A2M(Unknown Source:4071)
08-25 16:11:17.357 17517 17517 E AndroidRuntime:        at X.3aZ.get(Unknown Source:55)
08-25 16:11:17.357 17517 17517 E AndroidRuntime:        at X.3ab.get(:595754)
08-25 16:11:17.357 17517 17517 E AndroidRuntime:        at com.whatsapp.AbstractAppShellDelegate.onCreate(:483023)
08-25 16:11:17.357 17517 17517 E AndroidRuntime:        at X.0xe.onCreate(:122826)
08-25 16:11:17.357 17517 17517 E AndroidRuntime:        at X.1Dj.onCreate(:152041)
08-25 16:11:17.357 17517 17517 E AndroidRuntime:        at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1122)
08-25 16:11:17.357 17517 17517 E AndroidRuntime:        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6619)
08-25 16:11:17.357 17517 17517 E AndroidRuntime:        at android.app.ActivityThread.-wrap2(Unknown Source:0)
08-25 16:11:17.357 17517 17517 E AndroidRuntime:        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2066)
08-25 16:11:17.357 17517 17517 E AndroidRuntime:        at android.os.Handler.dispatchMessage(Handler.java:108)
08-25 16:11:17.357 17517 17517 E AndroidRuntime:        at android.os.Looper.loop(Looper.java:166)
08-25 16:11:17.357 17517 17517 E AndroidRuntime:        at android.app.ActivityThread.main(ActivityThread.java:7529)
08-25 16:11:17.357 17517 17517 E AndroidRuntime:        at java.lang.reflect.Method.invoke(Native Method)
08-25 16:11:17.357 17517 17517 E AndroidRuntime:        at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:245)
08-25 16:11:17.357 17517 17517 E AndroidRuntime:        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:921)
```

Thanks for your work! :)

**Comment by Schwartzblat** [comment]
Fixed. @jeverling @m00nty
